README.md
<!--
This README describes the package. If you publish this package to pub.dev,
this README's contents appear on the landing page for your package.

For information about how to write a good package README, see the guide for
[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).

For general information about developing packages, see the Dart guide for
[creating packages](https://dart.dev/guides/libraries/create-library-packages)
and the Flutter guide for
[developing packages and plugins](https://flutter.dev/developing-packages).
-->

# Rounded Bottom NavBar

First Open Source Flutter based Beautiful Material Design Bottom NavBar.(More designs coming soon.)

## Features

[*] Compatible on Android and iOS.

- [✓] Customizable textfield properties.
- [✓] Beautiful FontStyles
- [✓] User defined text controller
- [✓] Fully Responsive
- [✓] Loading indicator function (Coming Soon)

### Show some :heart: and star the repo to support the project

### Screenshots

<img src="ss1.jpg" height="300em" />&nbsp;&nbsp;&nbsp; <img src="ss2.jpg" height="300em" /> &nbsp;&nbsp;&nbsp;
<img src="ss3.jpg" height="300em" />

### Usage

[Example](https://github.com/darshn-n/rounded_bottom_navbar/blob/master/example/example_app.dart)

### Created & Maintained By

[Darshan N](https://github.com/darshn-n)

> If you found this project helpful or you learned something from the source code and want to thank me <darshan51081@gmail.com>

# License

    Copyright 2022 Darshan N

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at


    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.

## Getting started

For help getting started with Flutter, view our online
[documentation](https://flutter.dev/).

## Usage

- To add the following dependency to your [pubspec.yaml](https://github.com/darshn-n/rounded_bottom_navbar)

```yaml
dependencies:
  flutter:
    sdk: flutter
  rounded_bottom_navbar: Latest Version
```

```dart
       RoundedBottomNavBar(
        items: const <BottomNavigationBarItem>[
          BottomNavigationBarItem(
            icon: Icon(Icons.home),
            label: 'Home',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.business),
            label: 'Business',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.school),
            label: 'School',
          ),
        ],
        currentIndex: _selectedIndex,
        selectedItemColor: Colors.amber[800],
        onTap: _onItemTapped,
      ),
```

## Additional information

This package is Light-weight with every properties of Bottom Nav Bar.


CHANGELOG.md
## 0.0.1

* Initial release rounded_bottom_navbar.


lib/rounded_bottom_navbar.dart
library rounded_bottom_navbar;

import 'package:flutter/material.dart';

class RoundedBottomNavBar extends StatefulWidget {
  const RoundedBottomNavBar({super.key, required this.items});
  final List<BottomNavigationBarItem> items;

  @override
  State<RoundedBottomNavBar> createState() => _RoundedBottomNavBarState();
}

class _RoundedBottomNavBarState extends State<RoundedBottomNavBar> {
  @override
  Widget build(BuildContext context) {
    return Container(
      decoration: const BoxDecoration(
        borderRadius: BorderRadius.only(
          topRight: Radius.circular(
            30,
          ),
          topLeft: Radius.circular(
            30,
          ),
        ),
        boxShadow: [
          BoxShadow(
            color: Colors.black38,
            spreadRadius: 0,
            blurRadius: 10,
          ),
        ],
      ),
      child: ClipRRect(
        borderRadius: const BorderRadius.only(
          topLeft: Radius.circular(30.0),
          topRight: Radius.circular(30.0),
        ),
        child: BottomNavigationBar(
          items: widget.items,
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: const MyRoundedBottomNavBar(),
    );
  }
}

class MyRoundedBottomNavBar extends StatefulWidget {
  const MyRoundedBottomNavBar({super.key});

  static const TextStyle optionStyle =
      TextStyle(fontSize: 30, fontWeight: FontWeight.bold);
  static const List<Widget> _widgetOptions = <Widget>[
    Text(
      'Index 0: Home',
      style: optionStyle,
    ),
    Text(
      'Index 1: Business',
      style: optionStyle,
    ),
    Text(
      'Index 2: School',
      style: optionStyle,
    ),
  ];

  @override
  State<MyRoundedBottomNavBar> createState() => _MyRoundedBottomNavBarState();
}

class _MyRoundedBottomNavBarState extends State<MyRoundedBottomNavBar> {
  int _selectedIndex = 0;

  void _onItemTapped(int index) {
    setState(() {
      _selectedIndex = index;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('BottomNavigationBar Sample'),
      ),
      body: Center(
        child: MyRoundedBottomNavBar._widgetOptions.elementAt(_selectedIndex),
      ),
      bottomNavigationBar: BottomNavigationBar(
        items: const <BottomNavigationBarItem>[
          BottomNavigationBarItem(
            icon: Icon(Icons.home),
            label: 'Home',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.business),
            label: 'Business',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.school),
            label: 'School',
          ),
        ],
        currentIndex: _selectedIndex,
        selectedItemColor: Colors.amber[800],
        onTap: _onItemTapped,
      ),
    );
  }
}
